internal/operator/uninstall.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operator
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	v1 "github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/yaml"
+)
+
+type Uninstall struct {
+	config *Configuration
+
+	Package string
+}
+
+func NewUninstall(cfg *Configuration) *Uninstall {
+	return &Uninstall{
+		config: cfg,
+	}
+}
+
+func (u *Uninstall) Run(ctx context.Context) error {
+	subs := v1alpha1.SubscriptionList{}
+	if err := u.config.Client.List(ctx, &subs, client.InNamespace(u.config.Namespace)); err != nil {
+		return fmt.Errorf("list subscriptions: %v", err)
+	}
+
+	var sub *v1alpha1.Subscription
+	for _, s := range subs.Items {
+		if u.Package == s.Spec.Package {
+			sub = &s
```

Comment:
`sub` may be incorrect since `s` is an iterated value. See #2969.
```suggestion
	for i, s := range subs.Items {
		if u.Package == s.Spec.Package {
			sub = &subs.Items[i]
```

Reply:
I don't understand how this is different, but :man_shrugging: sounds good.